web/docs/data-model/prisma-file.md

```diff
@@ -0,0 +1,200 @@
+
+### The `datasource` block
+
+```prisma title="schema.prisma"
+datasource db {
+  provider = "postgresql"
+  url      = env("DATABASE_URL")
+}
+```
+
+Wasp will take the `datasource` you write and use it as-is.
+
+There are some rules you need to follow:
+- You can only use `"postgresql"` or `"sqlite"` as the `provider` because Wasp only supports PostgreSQL and SQLite databases. 
+- You must set the `url` field to `env("DATABASE_URL")` so that Wasp can work properly with your database.
+
+### The `generator` blocks
+
+```prisma title="schema.prisma"
+generator client {
+  provider = "prisma-client-js"
+}
+```
+
+Wasp requires that there is a `generator` block with `provider = "prisma-client-js"` in the `schema.prisma` file.
+
+You can add additional generators if you want to use them in your project.
+
+### The `model` blocks
+
+```prisma title="schema.prisma"
+model User {
+  id      Int        @id @default(autoincrement())
+  tasks   Task[]
+}
+
+model Task {
+  id          Int        @id @default(autoincrement())
+  description String
+  isDone      Boolean    @default(false)
+  user        User       @relation(fields: [userId], references: [id])
+  userId      Int
+}
+```
+
+You can define your models in any way you like, if it's valid Prisma schema code, it will work with Wasp.
+
+Some minor caveats to keep in mind:
+
+- Wasp doesn't support `/// comment` syntax in the `schema.prisma` file. We are tracking it [here](https://github.com/wasp-lang/wasp/issues/2132), let us know if this is something you need.
+- Wasp doesn't support MongoDB-specific [composite types](https://www.prisma.io/docs/orm/reference/prisma-schema-reference#type) in the `schema.prisma` file.
+
+## Prisma preview features
+
+Prisma is still in active development and some of its features are not yet stable. To use them, you have to enable them in the client `generator` block:
+
+```prisma title="schema.prisma"
+// ...
+
+generator client {
+  provider        = "prisma-client-js"
+  // highlight-next-line
+  previewFeatures = ["postgresqlExtensions"]
+}
+```
+
+<small>
+
+Read more about Prisma preview features in the [Prisma docs](https://www.prisma.io/docs/concepts/components/preview-features/client-preview-features).
+</small>
+
+### PostgreSQL Extensions
+
+PostgreSQL supports [extensions](https://www.postgresql.org/docs/current/contrib.html) that add additional functionality to the database. For example, the [hstore](https://www.postgresql.org/docs/13/hstore.html) extension adds support for storing key-value pairs in a single column.
+
+To use PostgreSQL extensions with Prisma, you have to enable them in the `datasource` block:
+
+```prisma title="schema.prisma"
+datasource db {
+  provider   = "postgresql"
+  url        = env("DATABASE_URL")
+  // highlight-next-line
+  extensions = [hstore(schema: "myHstoreSchema"), pg_trgm, postgis(version: "2.1")]
+}
+
+// ...
+```
```

Comment:
This is also strictly Prisma related. Although I guess it might be interesting since it is PostrgreSQL. But still, I would either remove it, or shorten it to be one or two sentences with a URL to prisma docs and that is it, like a super short FYI: "For how to use Postgre extensions with Prisma, check URL".